src/prefect/orion/models/artifacts.py

```diff
@@ -0,0 +1,50 @@
+from uuid import UUID
+
+import pendulum
+import sqlalchemy as sa
+
+from prefect.orion.database.dependencies import inject_db
+from prefect.orion.database.interface import OrionDBInterface
+from prefect.orion.schemas.core import Artifact
+
+
+@inject_db
+async def create_artifact(
+    session: sa.orm.Session, artifact: Artifact, db: OrionDBInterface, key: str = None
+):
+    now = pendulum.now("UTC")
+    artifact_id = artifact.id
+    insert_stmt = (await db.insert(db.Artifact)).values(
+        created=now,
+        updated=now,
```

Comment:
don't we normally rely on a trigger for managing the `updated` column, or am i off base?

Reply:
I think we use different strategies in various places, this is the pattern I believe we use if we have separate upsert logic if we're trying to overwrite an existing row